File: examples/ReportCreateExample.rb
require "json"
require "dropbox-sign"

Dropbox::Sign.configure do |config|
    config.username = "YOUR_API_KEY"
end

report_create_request = Dropbox::Sign::ReportCreateRequest.new
report_create_request.start_date = "09/01/2020"
report_create_request.end_date = "09/01/2020"
report_create_request.report_type = [
    "user_activity",
    "document_status",
]

begin
    response = Dropbox::Sign::ReportApi.new.report_create(
        report_create_request,
    )

    p response
rescue Dropbox::Sign::ApiError => e
    puts "Exception when calling ReportApi#report_create: #{e}"
end

File: examples/ReportCreateExample.php
<?php

namespace Dropbox\SignSandbox;

require_once __DIR__ . '/../vendor/autoload.php';

use SplFileObject;
use Dropbox;

$config = Dropbox\Sign\Configuration::getDefaultConfiguration();
$config->setUsername("YOUR_API_KEY");

$report_create_request = (new Dropbox\Sign\Model\ReportCreateRequest())
    ->setStartDate("09/01/2020")
    ->setEndDate("09/01/2020")
    ->setReportType([
        Dropbox\Sign\Model\ReportCreateRequest::REPORT_TYPE_USER_ACTIVITY,
        Dropbox\Sign\Model\ReportCreateRequest::REPORT_TYPE_DOCUMENT_STATUS,
    ]);

try {
    $response = (new Dropbox\Sign\Api\ReportApi(config: $config))->reportCreate(
        report_create_request: $report_create_request,
    );

    print_r($response);
} catch (Dropbox\Sign\ApiException $e) {
    echo "Exception when calling ReportApi#reportCreate: {$e->getMessage()}";
}
